src/MatrixFields/field_name_dict.jl

```diff
@@ -182,6 +182,16 @@ function Base.similar(dict::FieldNameDict)
     return FieldNameDict(keys(dict), entries)
 end
 
+function Base.zero(dict::FieldMatrix)
+    entries = unrolled_map(values(dict)) do entry
+        entry isa UniformScaling ? entry : zero(entry)
+    end
+    return FieldNameDict(keys(dict), entries)
+end
+
+Base.zero(dict::FieldNameDict) =
+    FieldNameDict(keys(dict), map(zero, values(dict)))
```

Comment:
```suggestion
function Base.zero(dict::FieldNameDict)
    entries = unrolled_map(values(dict)) do entry
        entry isa UniformScaling ? entry : zero(entry)
    end # TODO: Replace UniformScaling with mutable Ref to make this consistent
    return FieldNameDict(keys(dict), entries)
end
```

Reply:
It looks like the matrix-field method was needed after all: https://buildkite.com/clima/climaatmos-ci/builds/22412#01949b97-4f3c-477b-8ea5-b0a0e49491d7

Reply:
Oh, wait, that build shows that it's not using the latest ClimaCore.